### DISCUSSION TITLE
Upload image, get image_id and image details in one upload

### DISCUSSION THREAD

**Discussion by gpfreund** [exadel-inc/CompreFace#1129] [General]
Let me preface this that's I'm super-stoked with CompreFace and very much appricate all the work that must have gone into it...

Out application includes a selfie registration and validation of some aspects of the image (is it a person, yaw, ...). Because I  subsequently  need to use the same image for comparision with 2FA selfies, I want to store it in CompreFace. Is there any way to do this in one step. Right now we are uploading the image from our S3 bucket in order to get the image_id and then upload it  again to evaluate the image. This is obviously time consuming. (BTW: Ignore the details of the http calls, this our internal OkHttp wrapper). 
```
...
          stream = CoreAPI.downloadFile(s3Id, username);
          jResult = httpPostStream("recognition/faces", 
                                    stream, s3Id + ".jpg", "image/jpeg", jRequest,
                                    credentialUpload);
          if (jResult.getInt("status") >= HTTP.NO_ERROR) {
            return jResult;
          }
          stream = CoreAPI.downloadFile(s3Id, username);
          jResult1 = httpPostStream("recognition/recognize?face_plugins=age,gender,mask,pose",
                                    stream, s3Id + ".jpg", "image/jpeg", jRequest,
                                    credentialUpload);
          
          if (jResult1.getInt("status") >= HTTP.NO_ERROR) {
            return jResult;
          }
...
```

**Comment by pospielov** [comment]
I'm not sure I totally understand the case.
Here is what I see:
1. You download the image as a stream
2. You upload the image to CompreFace as an example to face collection
3. You download the (same?) image as a stream
4. You use a recognition service to recognize the face on it.
What I don't get:
1. Why do you need to recognize the image if you just saved it into the collection? CompreFace will return 100% similarity to it. 
2. For what purposes do you use image_id?
3. Do I understand correctly that the ultimate goal is to receive the result of `face_plugins` from the request that saves the example to face collection?
